### DISCUSSION TITLE
Hosting a Message Producer and Dispatcher under a same application

### DISCUSSION THREAD

**Discussion by honkuan86** [BrighterCommand/Brighter#1579] [General]
Currently a message producer and a service activator (dispatcher) can be hosted in different application. Is it possible that we can host both in a same application ?

**Comment by holytshirt** [comment]
Are you thinking about running the Dispatcher as a background tasks with hosted services in ASP.NET Core?

Have not tried it before but if you use the hosted service extension it should work.
https://github.com/BrighterCommand/Brighter/blob/master/src/Paramore.Brighter.ServiceActivator.Extensions.Hosting/ServiceActivatorHostedService.cs

**Reply by holytshirt** [reply]
You'll also have access to the producer that is registered as part of registering Service Activator

**Reply by honkuan86** [reply]
@holytshirt , I got it to work now after the hints by @preardon . Thanks.
